core/src/main/resources/mappings/scheduled-jobs.json

```diff
@@ -118,6 +118,46 @@
             }
           }
         },
+        "data_sources": {
+          "properties": {
+            "alerts_index": {
+              "type": "text",
+              "fields": {
+                "keyword": {
+                  "type": "keyword",
+                  "ignore_above": 256
+                }
+              }
+            },
+            "findings_index": {
+              "type": "text",
+              "fields": {
+                "keyword": {
+                  "type": "keyword",
+                  "ignore_above": 256
+                }
+              }
+            },
+            "query_index": {
+              "type": "text",
+              "fields": {
+                "keyword": {
+                  "type": "keyword",
+                  "ignore_above": 256
+                }
+              }
+            },
+            "query_index_mapping": {
+              "type": "text",
+              "fields": {
+                "keyword": {
+                  "type": "keyword",
+                  "ignore_above": 256
+                }
+              }
+            }
+          }
+        },
```

Comment:
What are the pros and cons of keeping as part of the metadata/config vs as a separate object part of the same index?

Reply:
Pros :
1. We load all metadata and configs at once for the data sources. If kept as a separate object, we would have to read another index and logically link it back to the monitor pojo. 
2. We can re-use data sources POJO for more complicated scenarios such as a possible composite monitor wanting to override a monitor's behaviour by linking monitor id with its own desired data source object to override behaviour of the monitor.